ibis/backends/flink/compiler/core.py

```diff
@@ -46,15 +49,53 @@ class FlinkCompiler(Compiler):
 
 def translate(op: ops.TableNode) -> str:
     # TODO(chloeh13q): support translation of non-select exprs (e.g. literals)
-    ast = FlinkCompiler.to_ast(op)
-    return translate_language(ast.queries[0])
+    return translate_op(op)
 
 
 @functools.singledispatch
-def translate_language(language: DML | DDL) -> str:
-    raise com.OperationNotDefinedError(f'No translation rule for {type(language)}')
-
-
-@translate_language.register(Select)
-def _select(language: Select):
-    return language.compile()
+def translate_op(op: ops.TableNode) -> str:
+    raise com.OperationNotDefinedError(f'No translation rule for {type(op)}')
+
+
+@translate_op.register(ops.Literal)
+def _literal(op: ops.Literal) -> str:
+    value = op.value
+    dtype = op.output_dtype
+
+    if dtype.is_boolean():
+        # TODO(chloeh13q): Flink supports a third boolean called "UNKNOWN"
+        return 'TRUE' if value else 'FALSE'
+    elif dtype.is_string():
+        quoted = value.replace("'", "''").replace("\\", "\\\\")
+        return f"'{quoted}'"
+    elif dtype.is_date():
+        if isinstance(value, datetime.date):
+            value = value.strftime('%Y-%m-%d')
+        return repr(value)
+    elif dtype.is_numeric():
+        if math.isnan(value):
+            raise ValueError("NaN is not supported in Flink SQL")
+        elif math.isinf(value):
+            raise ValueError("Infinity is not supported in Flink SQL")
+        return repr(value)
+    elif dtype.is_timestamp():
+        # TODO(chloeh13q): support timestamp with local timezone
+        if isinstance(value, datetime.datetime):
+            return 'TIMESTAMP ' + repr(value.strftime('%Y-%m-%d %H:%M:%S'))
+        raise NotImplementedError(f'No translation rule for timestamp {value}')
```

Comment:
Looking at the Clickhouse backend, it seems to just assume `.is_timestamp()` implies you can access attributes like `.microsecond`, so I wonder if this is necessary? Haven't checked other backend implementations (yet?).

Reply:
Also, this seems to be the main place you've errored if not `isinstance` for the timestamp checks; feels inconsistent, but I may be missing something.

Reply:
> I wonder if this is necessary

Is what necessary?

Reply:
> this seems to be the main place you've errored if not `isinstance` for the timestamp checks

I took the `isinstance` check from the base SQL implementation & BigQuery, the difference being that the other two implementations do not error out if it's not a `datetime` instance. I just chose to add an error here.